### PR TITLE
Add CountdownCalculator Feature

### PR DESCRIPTION
**Pull request by lineargraph**
This uses a predefined list of labels (because we all know how much of
an extremely good track record I have with those).

Amazing 24 hour and 12 hour time format support and even quick forge
works! We are truly living in the future!

### CODE REVIEW

**Comment by lineargraph** [comment]
nvm it doesnt work on cookies

**Comment by RayDeeUx** [comment]
actually it didnt work on anything bc nea messed up the regex

**Comment by RayDeeUx** [comment]
👍 all good to go

**Comment by lineargraph** [comment]
>edit: might revisit to reduce line count in the future but this works for now

stfu write the codeyyourself if you dont like mine okay

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/CountdownCalculator.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        Countdown("Time left:", "Ends at"),
+        Countdown("Event lasts for", "Ends at", isRelative = true),
+        Countdown("(§e", "Starts at"), // Calendar details
+    )
+
+
+    @SubscribeEvent
+    fun onTooltip(event: ItemTooltipEvent) {
+        if (!Utils.inSkyblock) return
+        val useFormatter = when(Skytils.config.showWhenCountdownEnds) {
```
Comment (My-Name-Is-Jeff) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/CountdownCalculator.kt]: code style

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/CountdownCalculator.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            0 -> return
```
Comment (My-Name-Is-Jeff) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/CountdownCalculator.kt]: just remove this case

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/CountdownCalculator.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    val formatter12h = DateTimeFormatter.ofPattern("EEEE, MMM d h:mm:ss a z")!!
+    val formatter24h = DateTimeFormatter.ofPattern("EEEE, MMM d HH:mm:ss z")!!
+
+    data class Countdown(
+        val match: String,
+        val label: String,
+        val isRelative: Boolean = false,
+    )
+
+    val countdownTypes = listOf(
```
Comment (My-Name-Is-Jeff) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/CountdownCalculator.kt]: enum maybe?

**Review by My-Name-Is-Jeff — APPROVED** [review]

**Pull request merged**